static/app/views/projectInstall/platformIntegrationSetup.tsx

```diff
@@ -1,172 +1,137 @@
-      installed: false,
-      integrations: {providers: []},
-    };
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    window.scrollTo(0, 0);
-
-    const {platform} = this.props;
-
-    // redirect if platform is not known.
-    if (!platform || platform.id === 'other') {
-      this.redirectToNeutralDocs();
-    }
-  }
-
-  get provider() {
-    const {providers} = this.state.integrations;
-    return providers.length ? providers[0] : null;
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization, integrationSlug} = this.props;
+}
 
-    if (!integrationSlug) {
-      return [];
+export function PlatformIntegrationSetup({
+  project,
+  platform,
+  onClickManualSetup,
+  integrationSlug,
+}: PlatformIntegrationSetupProps) {
+  const organization = useOrganization();
+  const [installed, setInstalled] = useState(false);
+  const navigate = useNavigate();
+
+  const {data: integrations} = useApiQuery<{providers: IntegrationProvider[]}>(
+    [
+      `/organizations/${organization.slug}/config/integrations/?provider_key=${integrationSlug}`,
+    ],
+    {
+      enabled: !!integrationSlug,
+      staleTime: 0,
     }
+  );
 
-    return [
-      [
-        'integrations',
-        `/organizations/${organization.slug}/config/integrations/?provider_key=${integrationSlug}`,
-      ],
-    ];
-  }
-
-  handleFullDocsClick = () => {
-    const {organization} = this.props;
-    trackAnalytics('growth.onboarding_view_full_docs', {organization});
-  };
-
-  redirectToNeutralDocs() {
-    const {organization, project} = this.props;
+  const provider = integrations?.providers.length ? integrations.providers[0] : null;
 
+  const redirectToNeutralDocs = useCallback(() => {
     if (!project) {
       return;
     }
 
     const url = `/organizations/${organization.slug}/projects/${project.slug}/getting-started/`;
 
-    browserHistory.push(normalizeUrl(url));
-  }
+    navigate(normalizeUrl(url));
+  }, [organization, project, navigate]);
 
-  handleAddIntegration = () => {
-    this.setState({installed: true});
-  };
-
-  trackSwitchToManual = () => {
-    const {onClickManualSetup, organization, integrationSlug} = this.props;
-    onClickManualSetup();
-    trackIntegrationAnalytics('integrations.switch_manual_sdk_setup', {
-      integration_type: 'first_party',
-      integration: integrationSlug,
-      view: 'project_creation',
-      organization,
-    });
-  };
-
-  render() {
-    const {organization, project, platform} = this.props;
-    const {installed} = this.state;
-    const provider = this.provider;
-
-    if (!provider || !platform || !project) {
-      return null;
+  useEffect(() => {
+    window.scrollTo(0, 0);
+    // redirect if platform is not known.
+    if (!platform || platform.id === 'other') {
+      redirectToNeutralDocs();
     }
+  }, [redirectToNeutralDocs, platform]);
+
+  if (!provider || !platform || !project) {
+    return null;
+  }
```

Comment:
I think there needs to be a loading state, the AsyncComponent before was handling this

Reply:
oh yes sure! always when I convert a component that was previously using `AsyncComponent` I forget about the loading state. Thanks 🙏